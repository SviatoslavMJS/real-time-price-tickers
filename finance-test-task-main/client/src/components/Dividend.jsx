import s from '../styles/Dividend.module.css';

function Dividend(props) {

    return (
        <div className={s.dividend}>
            {props.dividend}
        </div>
    )
}

export default Dividend;