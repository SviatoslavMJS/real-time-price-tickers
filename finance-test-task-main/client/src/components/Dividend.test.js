import renderer from 'react-test-renderer';
import Dividend from './Dividend';


test("On change prop dividend Dividend text must change from 666 to 000", () => {
    const component = renderer.create(
      <Dividend dividend={"666"} vector={true} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.dividend = "000";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.dividend = "666";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
});

test("On change prop vector Dividend class must change from green to red", () => {
    const component = renderer.create(
      <Dividend dividend={"666"} vector={true} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = false;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = true;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
});