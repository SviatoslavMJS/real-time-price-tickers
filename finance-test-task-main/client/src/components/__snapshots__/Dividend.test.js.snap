// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On change prop dividend Dividend text must change from 666 to 000 1`] = `
<div
  className="green"
>
  666
</div>
`;

exports[`On change prop dividend Dividend text must change from 666 to 000 2`] = `
<div
  className="green"
>
  666
</div>
`;

exports[`On change prop dividend Dividend text must change from 666 to 000 3`] = `
<div
  className="green"
>
  666
</div>
`;

exports[`On change prop vector Dividend class must change from green to red 1`] = `
<div
  className="green"
>
  666
</div>
`;

exports[`On change prop vector Dividend class must change from green to red 2`] = `
<div
  className="green"
>
  666
</div>
`;

exports[`On change prop vector Dividend class must change from green to red 3`] = `
<div
  className="green"
>
  666
</div>
`;
